From 8ae4bf1887a3f9de28f6342204138560006bb09f Mon Sep 17 00:00:00 2001
From: David Kershner <dkershner@habana.ai>
Date: Tue, 03 Jun 2025 05:35:26 +0300
Subject: [PATCH] [SW-180232] pldm: keep present state normal

With the addition of persisting the meta state of effecters, the
asic reset effecter was setting the present state to reset. This
caused the code during restore to issue another reset request,
so we would just constantly reset device. Change the pending_state
to reset, but leave the present_state as normal.

Change-Id: I1681a53caa9c43e3a3ebdf6421df61f35eb5d90a
Signed-off-by: David Kershner <dkershner@habana.ai>
---

diff --git a/arch/gaudi3/ras/pldm/pldm_effecter_handlers.c b/arch/gaudi3/ras/pldm/pldm_effecter_handlers.c
index 67f1a7c..d00c04f 100644
--- a/arch/gaudi3/ras/pldm/pldm_effecter_handlers.c
+++ b/arch/gaudi3/ras/pldm/pldm_effecter_handlers.c
@@ -230,7 +230,12 @@
 		/* Fall through */
 	case PLDM_STATE_SET_BOOT_RESTART_REQUEST_NORMAL:
 		reset_state = new_state;
-		composite_state->present_state = new_state;
+		composite_state->pending_state = new_state;
+		/* In case of Fall through, present state is persisted, setting
+		 * to RESTART REQUEST will cause constant restarts. Set to NORMAL
+		 * to only reset once.
+		 */
+		composite_state->present_state = PLDM_STATE_SET_BOOT_RESTART_REQUEST_NORMAL;
 		break;
 	default:
 		return -EINVAL;
@@ -245,7 +250,7 @@
 {
 	composite_state->op_state = PLDM_PMC_EFFECTER_OP_STATE_ENABLED_NO_UPDATE_PENDING;
 	composite_state->pending_state = reset_state;
-	composite_state->present_state = reset_state;
+	composite_state->present_state = PLDM_STATE_SET_BOOT_RESTART_REQUEST_NORMAL;
 
 	return 0;
 }
